app/src/main/java/com/kickstarter/services/KSApolloClientV2.kt

```diff
@@ -8,20 +8,62 @@ import com.kickstarter.models.Project
 import com.kickstarter.models.StoredCard
 import com.kickstarter.services.mutations.SavePaymentMethodData
 import com.kickstarter.services.transformers.encodeRelayId
+import com.kickstarter.services.transformers.projectTransformer
 import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
 import io.reactivex.subjects.PublishSubject
 import type.FlaggingKind
 
 interface ApolloClientTypeV2 {
+    fun getProject(project: Project): Observable<Project>
+    fun getProject(slug: String): Observable<Project>
     fun createSetupIntent(project: Project? = null): Observable<String>
     fun savePaymentMethod(savePaymentMethodData: SavePaymentMethodData): Observable<StoredCard>
     fun getStoredCards(): Observable<List<StoredCard>>
     fun deletePaymentSource(paymentSourceId: String): Observable<DeletePaymentSourceMutation.Data>
     fun createFlagging(project: Project? = null, details: String, flaggingKind: String): Observable<String>
     fun userPrivacy(): Observable<UserPrivacyQuery.Data>
+    fun watchProject(project: Project): Observable<Project>
+    fun unWatchProject(project: Project): Observable<Project>
 }
 
 class KSApolloClientV2(val service: ApolloClient) : ApolloClientTypeV2 {
+    override fun getProject(project: Project): Observable<Project> {
+        return getProject(project.slug() ?: "")
+    }
+    override fun getProject(slug: String): Observable<Project> {
+        return Observable.defer {
+            val ps = PublishSubject.create<Project>()
+            this.service.query(
+                FetchProjectQuery.builder()
+                    .slug(slug)
+                    .build()
+            ).enqueue(object : ApolloCall.Callback<FetchProjectQuery.Data>() {
+                override fun onFailure(e: ApolloException) {
+                    ps.onError(e)
+                }
+
+                override fun onResponse(response: Response<FetchProjectQuery.Data>) {
+                    if (response.hasErrors()) ps.onError(java.lang.Exception(response.errors?.first()?.message))
+                    else {
+                        response.data?.let { responseData ->
```

Comment:
If there is no response data available the observable is never completed, could you change the else branch to: 
```
                    if (response.hasErrors()) ps.onError(java.lang.Exception(response.errors?.first()?.message))
                    else {
                        response.data?.project()?.fragments()?.fullProject().let { project ->
                            ps.onNext(projectTransformer(project))
                        }
                    }
                    ps.onComplete()
```